packages/@sfx/core/src/core.ts

```diff
@@ -31,6 +37,13 @@ export default class Core {
    */
   directory: PluginDirectory = Object.create(null);
 
+  /**
+   * An object with plugin names mapped to the names of the plugins that
+   * depend on it. This dependency graph is represented as an adjacency
+   * list.
+   */
+  dependencyGraph: DependencyGraph = createDependencyGraph();
```

Comment:
High-level question: do we need a dependency graph if we already have a list of plugins, each plugin having its own list of plugins on which it depends? An alternative would be creating utilities for navigating our existing plugins and their dependencies without explicitly building a graph.

Reply:
For posterity: after some discussion, we concluded that there is a way to do it without building the graph. But also after discussion, we decided to leave this implementation for time's sake.

The alternate solution is as follows:

1. Create a hash (`Set`, `Object`, `Map`; something with O(1) lookup) from the plugins to remove.
2. For each plugin in the registry, excluding the ones to be removed, check that it has none of the plugins to remove listed in its dependencies. If it does, add the plugin's name to a structure that associates the name with the dependency.
3. If the dependencies are clean, the check passes. If not, throw with an appropriate error with a message that includes the dependencies and its dependers.

Worst case, this solution is O(n^2), but on average, it will probably be O(n).